flow/record/base.py

```diff
@@ -756,6 +756,30 @@ def fieldtype(clspath):
     return t
 
 
+@functools.lru_cache(maxsize=4069)
+def merge_record_descriptors(descriptors, replace=False, name=None):
+    """Create a newly merged RecordDescriptor from a list of RecordDescriptors.
+    This function uses a cache to avoid creating the same descriptor multiple times.
+
+    Duplicate fields are ignored in `descriptors` unless `replace=True`.
+
+    Args:
+        descriptors (Tuple[RecordDescriptor]): Tuple of RecordDescriptors we use for extending/replacing.
+        replace (bool): if `True`, it will replace existing field names, eg: last descriptor always wins.
```

Comment:
```suggestion
        replace (bool): if ``True``, it will replace existing field names. Last descriptor always wins.
```
It's not really an example if it's a rule.